api/src/main/java/org/pmiops/workbench/workspaces/WorkspacesController.java

```diff
@@ -558,6 +521,33 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
             .workspace(workspaceMapper.toApiWorkspace(savedWorkspace, toFcWorkspace)));
   }
 
+  // A retry period is needed because the permission to copy files into the cloned workspace is not granted transactionally
+  private void copyBlobWithRetries(String bucketName, Blob b, int retryPeriodSeconds) {
+    final int SLEEP_INTERVAL_SECONDS = 5;
+
+    try {
+      cloudStorageService.copyBlob(b.getBlobId(), BlobId.of(bucketName, b.getName()));
+    } catch (StorageException e) {
```

Comment:
Do you want to inspect the  `StorageException` you get back to make sure it's really an access problem?

Reply:
I think I'm ok with casting a wider net by catching on `StorageException`. If its due to some other problem, we'll see the exception get bubbled up after the retries are over.

Reply:
Oh right because you rethrow. Good deal.